src/runtime/thread_storage_scope.h

```diff
@@ -194,13 +198,17 @@ struct ThreadWorkLoad {
   inline size_t grid_dim(size_t i) const { return work_size[i]; }
 };
 /*! \brief Thread axis configuration */
-class ThreadAxisConfig {
+class LaunchParamConfig {
  public:
-  void Init(size_t base, const std::vector<std::string>& thread_axis_tags) {
+  void Init(size_t base, const std::vector<std::string>& launch_param_tags) {
     base_ = base;
     std::vector<bool> filled(6, false);
-    for (size_t i = 0; i < thread_axis_tags.size(); ++i) {
-      const std::string& tag = thread_axis_tags[i];
+    for (size_t i = 0; i < launch_param_tags.size(); ++i) {
+      const std::string& tag = launch_param_tags[i];
+      if (tag == kUseDynamicSharedMemoryTag) {
```

Comment:
let us always assert this is the last tag that follows the arg_index map. Since it can be strange if the tag appears in the middle.

We can also handle a generalized case where the arg indicates the location of the parameter. In that case if the tag is 
```[dyn_mem, threadIdx.x]```, then `x.values[base_+0]` would correspond to the dynamic shared mem size

Reply:
Added an assert